### ISSUE TITLE
Feature request: add 'changelog' metadata field

### ISSUE THREAD

**Issue by billgeo** [linz/stac#183]
**Is your feature request related to a problem? Please describe.**

So that I can understand the changes that have occurred to the dataset over time, as a Data User, I want to see a changelog for the dataset/collection.

**Describe the solution you'd like**

Add an optional linz:changelog metadata field to the collection metadata

Location: collection top level field metadata
Name: linz:changelog
Description: 

> Contains a log of changes that have occurred within the data collection. The Added, Modified, and Deleted counts fields are recommended and should be automatically populated from the dataset change. These counts represent the following for given data content types: 
>    Vector layers - Feature counts
>    Tables - Row counts
>     Rasters - Tile file counts
>     Point Clouds - Point counts.
>     Images - Image file counts"

Type: "Required for 'non-legacy' datasets"
Restrictions: An object that must have 'date' (UTC datetime), 'version' (string), 'message' (string); and can include 'added' (integer), 'modified' (integer), `deleted' (integer). Changelog history should ideally be ordered newest to oldest.

Example

```
Date: 2020-04-22T02:17:24.900422Z
Version: "1.3.0"
Message:  "Replaced tile BG37_5000_1009 to include nodata region.

A LDS customer noticed that tile BG37_5000_1009 had black regions on the outer edges of the image. The tile was reprocessed using GDAL to set the black regions to a nodata value to match the other tiles nodata values within the dataset.

Refer to issue https://jira.linz.govt.nz/browse/TODATA-654"
Added: 1
Modified: 0
Deleted: 0
```

**Describe alternatives you've considered**

**Additional context**

**Comment by l0b0** [comment]
What does `non-legacy` mean in this case? If it's not something which we're currently enforcing in the schema we should just make this field optional in this issue.

**Comment by billgeo** [comment]
Yes. I think it means some dataset that is old and doesn't have any changelog information and may only ever have one version. So yes, should be optional in the JSON schema I think.

**Comment by billgeo** [comment]
Maybe in the description it might make more sense to say 'Required for new datasets'

**Comment by l0b0** [comment]
> Maybe in the description it might make more sense to say 'Required for new datasets'

Isn't that tautological, since only new datasets will be referring to the latest version of the LINZ extension?

**Comment by l0b0** [comment]
I presume this is meant to be used for _non-atomic_ updates. For example, we don't want a changelog entry per file, but rather a semantically related change like re-aligning some geospatial reference across all assets or adding a field to all relevant metadata files. Do we want some guidelines along that line, to make sure we don't get spammed with changelog entries?

**Comment by l0b0** [comment]
Do we need some guidelines on the value for `date`? As a programmer, I envision a process like this one:

1. Someone discovers a change is needed (new data arrives, old data needs to be corrected, or something else).
2. Someone initiates the change process. This could take anything from minutes to months.
3. Eventually someone commits/pushes/saves the final change to the assets and metadata.

I would expect that the `date` value corresponds to the time of either #3 above or some time after that when someone updates the `date` value. What do you think?

**Comment by l0b0** [comment]
Should `message` maybe be `description`? I get connotations of *sending* this to someone if it's called "message", but that might just be me :).

**Comment by billgeo** [comment]
> Should message maybe be description

Yes. I think that's consistent iwth other stac metadata. Let's propose that.

**Comment by billgeo** [comment]
Victor is saying this is quite straight forward but may inlcude changes from feedback and rebasing etc.

**Comment by m-mohr** [comment]
I actually like this idea and could also see this being part of the version extension, although a changelog may only need to be linked to. So adding a rel type for changelogs might be they way I'd propose it in the version extension.
